src/libraries/pretest.proj

```diff
@@ -68,11 +68,31 @@
              ContinueOnError="ErrorAndContinue" />
   </Target>
 
+  <Target Name="RunCrossgen2OnHost" Condition="'$(Crossgen2Host)' == 'true'">
```

Comment:
Why Pretest and not a step in each libraries project file?  This makes things harder on inner loop if you care about crossgen2.

Reply:
With #31712, we are going to be illink trimming all the assemblies in the shared framework together in a "post build" step.

We can't crossgen in a step in each libraries project file because that will run before we run the linker. We will need to run crossgen on the output of the linker.